pyiron_base/jobs/job/runfunction.py

```diff
@@ -71,16 +71,11 @@ def write_input_files_from_input_dict(input_dict: dict, working_directory: str):
         input_dict (dict): hierarchical input dictionary with files_to_create and files_to_copy.
         working_directory (str): path to the working directory
     """
-    if len(os.listdir(working_directory)) == 0:
-        for file_name, content in input_dict["files_to_create"].items():
-            with open(os.path.join(working_directory, file_name), "w") as f:
-                f.writelines(content)
-        for file_name, source in input_dict["files_to_copy"].items():
-            shutil.copy(source, os.path.join(working_directory, file_name))
-    else:
-        state.logger.info(
-            "The working directory is not empty, assuming that input has already been written."
-        )
+    for file_name, content in input_dict["files_to_create"].items():
```

Comment:
```suggestion
    if len(os.listdir(working_directory)) > 0:
        state.logger.info(
            "The working directory is not empty, assuming that input has already been written."
        )
    for file_name, content in input_dict["files_to_create"].items():
```

Reply:
Formatting looks weird, but I wanted to keep the warning.

Reply:
This would be logged for every job sent to the cluster, i.e. most.  That's a bit excessive.

Reply:
No, the previous PR was a workaround, this one restores core functionality.  I've anyway updated the check to only emit the warning and skip the function body again when any of the files about to be created already exist.